docs/cypress/e2e/links.cy.ts

```diff
@@ -10,34 +10,43 @@ type EvtName =
   | 'REQUESTING'
   | 'RETURNING'
   | 'NO_HREF';
-let allLinks: string[] = [];
-const numberOfLinks = 121;
-const requestedLinks: Set<string> = new Set();
 
-before(() => {
-  cy.task('readSitemapLinks').then((links: string[]) => {
-    allLinks = allLinks.concat(links);
-  });
-});
+describe(`All pages on Sitemap`, () => {
+  let allLinks = [];
+  const requestedLinks: Set<string> = new Set();
 
-describe('Local Sitemap', () => {
-  it('should have 119 links', () => {
-    expect(allLinks.length).to.eq(numberOfLinks);
+  before(() => {
+    cy.request('sitemap.xml').then((response) => {
+      allLinks = Cypress.$(response.body)
+        .find('loc')
+        .toArray()
+        .map((el) => el.innerText)
+        .map((link) =>
+          link
+            .replace(`${BASE_URL}/`, '')
+            .replace('https://www.dev.ui.docs.amplify.aws/', '')
+            .replace('https://ui.docs.amplify.aws/', '')
+        );
+    });
   });
-});
 
-describe(`All pages on Sitemap`, () => {
-  it.each(numberOfLinks)('all links on page %k should work', (i: string) => {
-    const link = allLinks[i];
-    cy.task('log', `🧪[TESTING...] page ${BASE_URL}/${link}`);
-    cy.visit(link || '/');
-    cy.get('a').each((el) => hrefWorks(el, link));
+  it('should succesfully load each url in the sitemap', () => {
+    allLinks.forEach((link, idx) => {
+      cy.task('log', `🧪[TESTING...] page #${idx} ${BASE_URL}/${link}`);
+      cy.visit(link || '/');
+      cy.get('a').each((el) => hrefWorks(el, link, allLinks, requestedLinks));
```

Comment:
If I understand this right, what we are doing here is going through each page in the sitemap list, retrieving all links on each page and testing them. We have quite a bit of logic in the `hrefWorks` fn to exclude links we've already tested as being part of sitemap, which makes this a bit fragil and almost requires a test for our test. What do you think about simplifying?

I've analyzed the links on most of our pages and found a couple patterns. We have:
* external links [github, discord, aws, etc] (rule: doesn't have our baseurl)
* internal links - same as the ones in sitemap but pointing to a section (rule: contains #)
* links that are in sitemap so we don't want to re-test them

So I was thinking that instead of line 37 we can break down the links testing like so:

```
cy.get("a:not([href^='${DOCS_URL}']), a[href*='#']").each ... call simplified fn that only does cy.request, checks status, logs 
```

Reply:
I like this idea 👍

Reply:
Updated, but we still need to keep the `hrefWorks` function because we still need to check if it's in `requestedLinks`

Reply:
I thought about it again. We need to verify if the href in `a` tag is in sitemap, not outdated, so we can't skip them.

I'll revert the commit.

Reply:
I see. Can we not add a separate: 
```
cy.get("a[href^='${DOCS_URL}']).each ... 
```
following `cy.get("a:not([href^='${DOCS_URL}']), a[href*='#']").each` ?

this way we also test the links that contain '#' which now we are missing as per the TODO

Reply:
The major reason why we are not testing `#` is to make less request so that we can save some memory 😂

Reply:
I see! I separated them by `cy.get(`a[href^='/']`)` and `cy.get(`a:not([href^='/']), a[href*='#']`)` for internal and external links. It does make the `if...else` shorter now